node/router/src/handshake.rs

```diff
@@ -296,7 +296,7 @@ impl<N: Network> Router<N> {
         // Finalize the connecting peer information.
         self.connecting_peers.lock().insert(peer_ip, Some(Peer::new(peer_ip, peer_addr, &peer_request)));
         // Adds a bidirectional map between the listener address and (ambiguous) peer address.
-        self.resolver.insert_peer(peer_ip, peer_addr);
```

Comment:
Same comment as for `remove_connected_peer`: Would it make sense to hold the lock throughout the function, so we update `connected_peers` and `candidate_peers` atomically?

Note, here we modify the two in a different order. So you would need to lock the resolver before changing connected peers to prevent deadlocks.